.github/workflows/upload-test-seed.yml

```diff
@@ -0,0 +1,94 @@
+
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Comment "Upload In Progress"
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            const commentBody =
+            `## 🔄 Test Seed Upload In Progress
+
+            A new seed file is currently being uploaded for this pull request.
+
+            ### What's Next?
+
+            - The upload process typically takes a few minutes.
+            - Once the upload is complete, this comment will provide the seed details and testing instructions.
+            `
+            const comment = require('.github/workflows/scripts/comment.js')
+            await comment(github, context, commentBody)
+
+      - name: Install dependencies and run tests
+        run: |
+          npm install
+          npm run typecheck:scripts
+          npm run build:proto
+          npm run typecheck
+          npm run test
+
+      - name: Generate seed
+        run: |
+          npm run seed_tools -- create_seed studies seed.bin
+
+      - name: Upload seed
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_PRODUCTION_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_PRODUCTION_SECRET_ACCESS_KEY }}
+          AWS_REGION: us-west-2
+          CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.CLOUDFRONT_DISTRIBUTION_ID }}
+        run: |
+          gzip -c seed.bin | aws s3 cp - "s3://brave-production-griffin-origin/$REMOTE_SEED_PATH" \
+            --content-type application/octet-stream \
+            --content-encoding gzip
+
+          INVALIDATION_ID=$(aws cloudfront create-invalidation --distribution-id "$CLOUDFRONT_DISTRIBUTION_ID" --paths "/$REMOTE_SEED_PATH" --query 'Invalidation.Id' --output text)
+          aws cloudfront wait invalidation-completed --distribution-id "$CLOUDFRONT_DISTRIBUTION_ID" --id "$INVALIDATION_ID"
+
+      - name: Comment "Upload Successful"
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            const fs = require('fs');
+            const variationsServerURL = `https://griffin.brave.com/${process.env.REMOTE_SEED_PATH}`;
+            const serialNumberContent = fs.readFileSync('serialnumber', 'utf8');
+            const commentBody =
+            `## 🚀 Test Seed Upload Successful
+
+            To test the new seed, launch the browser with the following command line:
+            \`\`\`
+            --accept-empty-variations-seed-signature --variations-server-url=${variationsServerURL}
+            \`\`\`
+
+            #### Seed Details
+            - Serial Number: \`${serialNumberContent}\`
+            - Uploaded: \`${new Date().toISOString()}\`
+            `
+            const comment = require('.github/workflows/scripts/comment.js')
+            await comment(github, context, commentBody)
+
+      - name: Comment "Upload Failed"
```

Comment:
I would prefer to move this up and rename to "Upload skipped due to a build/lint/test failure".

Not sure that we should make a fallback for s3 uploading / invalidation, because it's a very small chance of failure here. IMHO, if it happens it's enough to have "Upload In Progress" comment and the failed job in the list.